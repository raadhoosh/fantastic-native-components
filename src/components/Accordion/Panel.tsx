import * as React from "react";
import { StyledHeader, StyledHeaderText, StyledContent, StyledBody } from "./Panel.style";
import { ViewStyle, ModalProps, View, Modal, TouchableOpacity } from "react-native";
import { Icon, Button } from "../../components";
import { Theme } from "..";
type IStyle=  ViewStyle | object | Array<ViewStyle>;
interface IProps {
    header?: JSX.Element | JSX.Element[];
    body?: JSX.Element | JSX.Element[];
    styledContent?: IStyle;
    children?: JSX.Element | JSX.Element[] | string;
    icon?: JSX.Element;
    visible?: boolean;
    primary?: boolean;
    secondary?: boolean;
    success?: boolean;
    info?: boolean;
    warning?: boolean;
    danger?: boolean;
    light?: boolean;
    dark?: boolean;
    backgroundColor?: string;
    title?: string;
    toggle?: () => void;
}

class Panel extends React.PureComponent<IProps> {
    renderHeader = () => {
        const { header } = this.props;
        if (header) {
            return header;
        }
        return (<StyledHeader onPress={this.props.toggle} {...this.props}>
            <StyledHeaderText>
                {this.props.title}
            </StyledHeaderText>
            {this.props.icon}
        </StyledHeader >);
    }
    renderBody = () => {
        const { visible, body } = this.props;
        if (visible) {
            if (body) {
                return body;
            }
            return ((<StyledBody {...this.props}>
                {this.props.children}
            </StyledBody>));
        }
        return null;
    }
    render() {
        const { styledContent } = this.props;
        let style: IStyle = { flex: 1, marginTop: 5 };
        if (styledContent) {
            style = styledContent;
        }
        return (
            <StyledContent style={style}>
                {this.renderHeader()}
                {this.renderBody()}
            </StyledContent>
        );
    }

}

export default Panel;
