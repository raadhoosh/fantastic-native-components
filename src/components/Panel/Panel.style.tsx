import styled from "styled-components";
import { ViewProps, View, Text, TouchableOpacity } from "react-native";
import { Theme } from "..";

interface IProps extends ViewProps {
    primary?: boolean;
    secondary?: boolean;
    success?: boolean;
    info?: boolean;
    warning?: boolean;
    danger?: boolean;
    light?: boolean;
    dark?: boolean;
    inverse?: boolean;
    backgroundColor?: string;
    theme?: Theme;
}

function getColor(props: IProps) {

    let TextInputColor = props.theme && props.theme.button.color ? props.theme.button.color : "#000";
    const inColor = (props.primary && "primary")
        || (props.secondary && "secondary")
        || (props.success && "success")
        || (props.info && "info")
        || (props.warning && "warning")
        || (props.danger && "danger");

    if (typeof (inColor) === "string" && props.theme) {
        TextInputColor = props.theme[inColor].main;
    }

    return TextInputColor;
}

const color = (props: IProps) => getColor(props);

const StyledContent = styled(View)`
    flex: 1;
    padding-left: 8;
    padding-right: 8;
`;

const StyledHeader = styled(TouchableOpacity)`
    background-color: ${(props: IProps) => props.backgroundColor ? props.backgroundColor : (props.inverse ? "#fff" : color)};
    padding: 10px;
    display: flex;
    flex-direction: row;
`;

const StyledHeaderText = styled(Text)`
    color: #fff;
    font-size: 14px;
    width: 95%;
`;

const StyledBody = styled(View)`
    display:flex;
    text-align: center;
    background-color: ${(props: IProps) => props.backgroundColor ? props.backgroundColor : (props.inverse ? color : "#fff")};
    width: auto;
    padding-top: 5;
    padding-bottom: 5;
    padding-left: 10;
    padding-right: 10;
    border: 1px;
    border-color: ${(props: IProps) => props.backgroundColor ? props.backgroundColor : (props.inverse ? "#fff" : color)};
`;

export { StyledHeader, StyledContent, StyledBody, StyledHeaderText };