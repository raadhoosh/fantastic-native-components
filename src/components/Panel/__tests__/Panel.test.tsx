import Panel from "..";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { View, Text } from "react-native";

describe("Panel", () => {
    describe("rendering", () => {
        let wrapper: ShallowWrapper;
        beforeEach(() => {
            wrapper = shallow(<Panel visible={true} primary title="panel">
                <View>
                    <Text>Test</Text>
                </View>
            </Panel>);
        });
        it("renders", () => {
            expect(wrapper.exists()).toBe(true);
        });
        it("toMatchSnapshot", () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});