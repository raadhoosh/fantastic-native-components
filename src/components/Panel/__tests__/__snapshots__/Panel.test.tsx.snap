// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel rendering toMatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel
    primary={true}
    title="panel"
    visible={true}
  >
    <Component>
      <Component>
        Test
      </Component>
    </Component>
  </Panel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          primary={true}
          title="panel"
          visible={true}
        >
          <ForwardRef>
            panel
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef
          primary={true}
          title="panel"
          visible={true}
        >
          <Component>
            <Component>
              Test
            </Component>
          </Component>
        </ForwardRef>,
      ],
      "style": Object {
        "flex": 1,
        "marginTop": 5,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              panel
            </ForwardRef>,
            undefined,
          ],
          "onPress": undefined,
          "primary": true,
          "title": "panel",
          "visible": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "panel",
            },
            "ref": null,
            "rendered": "panel",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
    color: #fff;
    font-size: 14px;
    width: 95%;
",
                ],
              },
              "displayName": "Styled(Component)",
              "propTypes": Object {
                "accessible": [Function],
                "adjustsFontSizeToFit": [Function],
                "allowFontScaling": [Function],
                "disabled": [Function],
                "ellipsizeMode": [Function],
                "minimumFontScale": [Function],
                "nativeID": [Function],
                "numberOfLines": [Function],
                "onLayout": [Function],
                "onLongPress": [Function],
                "onPress": [Function],
                "pressRetentionOffset": [Function],
                "selectable": [Function],
                "selectionColor": [Function],
                "style": [Function],
                "suppressHighlighting": [Function],
                "testID": [Function],
                "textBreakStrategy": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          undefined,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
    background-color: ",
              [Function],
              ";
    padding: 10px;
    display: flex;
    flex-direction: row;
",
            ],
          },
          "displayName": "Styled(TouchableOpacity)",
          "getDefaultProps": [Function],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component>
            <Component>
              Test
            </Component>
          </Component>,
          "primary": true,
          "title": "panel",
          "visible": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component>
              Test
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
    display:flex;
    text-align: center;
    background-color: ",
              [Function],
              ";
    width: auto;
    padding-top: 5;
    padding-bottom: 5;
    padding-left: 10;
    padding-right: 10;
    border: 1px;
    border-color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(Component)",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    flex: 1;
    padding-left: 8;
    padding-right: 8;
",
        ],
      },
      "displayName": "Styled(Component)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            primary={true}
            title="panel"
            visible={true}
          >
            <ForwardRef>
              panel
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef
            primary={true}
            title="panel"
            visible={true}
          >
            <Component>
              <Component>
                Test
              </Component>
            </Component>
          </ForwardRef>,
        ],
        "style": Object {
          "flex": 1,
          "marginTop": 5,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                panel
              </ForwardRef>,
              undefined,
            ],
            "onPress": undefined,
            "primary": true,
            "title": "panel",
            "visible": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "panel",
              },
              "ref": null,
              "rendered": "panel",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "rules": Array [
                    "
    color: #fff;
    font-size: 14px;
    width: 95%;
",
                  ],
                },
                "displayName": "Styled(Component)",
                "propTypes": Object {
                  "accessible": [Function],
                  "adjustsFontSizeToFit": [Function],
                  "allowFontScaling": [Function],
                  "disabled": [Function],
                  "ellipsizeMode": [Function],
                  "minimumFontScale": [Function],
                  "nativeID": [Function],
                  "numberOfLines": [Function],
                  "onLayout": [Function],
                  "onLongPress": [Function],
                  "onPress": [Function],
                  "pressRetentionOffset": [Function],
                  "selectable": [Function],
                  "selectionColor": [Function],
                  "style": [Function],
                  "suppressHighlighting": [Function],
                  "testID": [Function],
                  "textBreakStrategy": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            undefined,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
    background-color: ",
                [Function],
                ";
    padding: 10px;
    display: flex;
    flex-direction: row;
",
              ],
            },
            "displayName": "Styled(TouchableOpacity)",
            "getDefaultProps": [Function],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component>
              <Component>
                Test
              </Component>
            </Component>,
            "primary": true,
            "title": "panel",
            "visible": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Component>
                Test
              </Component>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
    display:flex;
    text-align: center;
    background-color: ",
                [Function],
                ";
    width: auto;
    padding-top: 5;
    padding-bottom: 5;
    padding-left: 10;
    padding-right: 10;
    border: 1px;
    border-color: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(Component)",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    flex: 1;
    padding-left: 8;
    padding-right: 8;
",
          ],
        },
        "displayName": "Styled(Component)",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
